pkg/apiserver/rest/dualwriter_mode2.go

```diff
@@ -9,25 +9,34 @@ import (
 	metainternalversion "k8s.io/apimachinery/pkg/apis/meta/internalversion"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apiserver/pkg/endpoints/request"
 	"k8s.io/apiserver/pkg/registry/rest"
 	"k8s.io/klog/v2"
 
+	"github.com/grafana/grafana/pkg/apimachinery/identity"
 	"github.com/grafana/grafana/pkg/apimachinery/utils"
+	grafanarequest "github.com/grafana/grafana/pkg/services/apiserver/endpoints/request"
 )
 
 type DualWriterMode2 struct {
 	Storage Storage
 	Legacy  LegacyStorage
 	*dualWriterMetrics
-	Log klog.Logger
+	Log        klog.Logger
+	Group      string
```

Comment:
Minor detail: it seems that Group, Resource, Namespacer are only required in order to construct `RequestInfo{}`. What if we had a RequestInfo field instead?

Reply:
That's a great point! I will modify the PR to reflect that change